crates/swc_ecma_transforms_base/src/helpers/_apply_decs_2203_r.js

```diff
@@ -476,6 +483,14 @@ function applyDecs2203RFactory() {
         // so we don't have to return an empty array here.
     }
 
+    function defineMetadata(Class, metadata) {
```

Comment:
Did you write this by yourself? I want to know if there was a reference implementation

Reply:
The js part is almost copied and adapted from babel implementation, while the Rust part obviously not
Tests are copied from babel